respringless_tweaks.md

```diff
@@ -0,0 +1,152 @@
+# How do you create a respringless tweak?
+
+* It's actually really easy if you understand the basics of it. What I'll make is a simple tweak that adds a blur to the LockScreen, with the option to change the its alpha/intensity on the fly. Think about it, how could that happen without a respring? Simple, Preferences needs to communicate with SpringBoard to notify that the blur value has been changed, so that it can refresh the UI. The `NSNotificationCenter` Foundation API makes that really straightforward.
+I won't explain what `NSNotificationCenter` does or how it works in too much detail because if you've got this far then you should already know the basics of it, if not then I'd suggest reading its documentation.
+
+## Tweak
+
+Open your terminal and create a tweak project, your `Tweak.x` should look like this:
+
+```objc
+#import "Tweah.h"
+
+
+static float blurIntensity;
+
+static void loadPrefs(void) {
+
+    NSUserDefaults *prefs = [[NSUserDefaults alloc] initWithSuiteName: kSuiteName];
+    blurIntensity = [prefs objectForKey:@"blurIntensity"] ? [prefs floatForKey:@"blurIntensity"] : 0.85;
+
+}
+
+static _UIBackdropView *blurView;
+
+%hook CSCoverSheetViewController
+
+%new
+
+- (void)setupBlur {
+
+    _UIBackdropViewSettings *settings = [_UIBackdropViewSettings settingsForStyle:2];
+
+    blurView = [[_UIBackdropView alloc] initWithFrame:CGRectZero autosizesToFitSuperview:YES settings:settings];
+    blurView.alpha = blurIntensity;
+    [self.view insertSubview:blurView atIndex:0];   
+
+}
+
+%new
+
+- (void)updateBlurIntensity {
+
+    loadPrefs();
+    blurView.alpha = blurIntensity;
+
+}
+
+- (void)viewDidLoad {
+
+    %orig;
+    [self setupBlur];
+
+    [NSDistributedNotificationCenter.defaultCenter addObserver:self selector:@selector(updateBlurIntensity) name:RespringlessTweakDidUpdateBlurIntensityNotification object: nil];
+
+}
+
+%end
+
+%ctor {
+    loadPrefs();
+}
+```
+
+Then your `Tweak.h` should have:
+
+```objc
+@import UIKit;
+#import "Common.h"
+
+
+@interface _UIBackdropViewSettings : NSObject
++ (id)settingsForStyle:(NSInteger)style;
+@end
+
+
+@interface _UIBackdropView : UIView
+- (id)initWithFrame:(CGRect)frame autosizesToFitSuperview:(BOOL)autosizes settings:(_UIBackdropViewSettings *)settings;
+@end
+
+
+@interface CSCoverSheetViewController: UIViewController
+- (void)setupBlur;
+@end
+```
+
+Finally, `Common.h`
+
+```objc
+static NSString *const kSuiteName = @"me.luki.respringlesstweakprefs";
+
+static NSNotificationName const RespringlessTweakDidUpdateBlurIntensityNotification = @"RespringlessTweakDidUpdateBlurIntensityNotification";
```

Comment:
Using a more specific notification name would be nicer, for example `com.example.respringlesstweakprefs/DidUpdateBlurIntensityNotification`. Also I think removing the constant is best